src/snowflake/snowpark/dataframe.py

```diff
@@ -2742,8 +2748,22 @@ def union_by_name(self, other: "DataFrame", _emit_ast: bool = True) -> "DataFram
             -------------
             <BLANKLINE>
 
+        Example::
+
+            >>> df1 = session.create_dataframe([[1, 2]], schema=["a", "b"])
+            >>> df2 = session.create_dataframe([[2, 1, 3]], schema=["b", "a", "c"])
+            >>> df1.union_by_name(df2, allow_missing_columns=True).show()
+            --------------------
+            |"A"  |"B"  |"C"   |
+            --------------------
+            |1    |2    |NULL  |
+            |1    |2    |3     |
+            --------------------
+            <BLANKLINE>
+
         Args:
             other: the other :class:`DataFrame` that contains the rows to include.
+            allow_missing_columns: When true includes missing columns in the final result. Missing values are Null filled. Default False.
```

Comment:
can you file a ticket under this: https://snowflakecomputing.atlassian.net/browse/SNOW-1916219 to get this option added to ast